### DISCUSSION TITLE
Followed the tutorial, ended with gibberish at worst, bad at best

### DISCUSSION THREAD

**Discussion by Vektor369** [JarodMica/ai-voice-cloning#142] [Q&A]
I followed the video "How to Clone Most Languages Using Tortoise TTS". The best I was able to get was by turning off "Hifigan" and selecting the tortoise tokenizer in the "Settings" tab, and using the "High Quality" Preset when generating. But then, of course, it takes 3 minutes to generate 8 words, and still sounded horrible.

Any help would be appreciated. Thanks in advance!

My training settings were:
(Over 5 minutes of high-quality audio)
epochs: 50
learning rate: 0.00001
Mel LR Ratio: 1
Text LR Ratio: 0.01
Learning Rate Scheme: Cos. Annealing
Learning Rate Restarts: 4
Batch Size: 8
Gradient Accumulation Size: 2
Save Frequency: 50
Validation Frequency: 50
Half Precision: no
BitsAndBytes: yes
Validation Enabled: no
Worker Processes: 2
lines: 134
lines/batches: 17
steps: 850

My results stats were:
Time taken: 1hr 34min
Epoch time: 110.690/s
IT time: 7.882/s
LR: 1.25E-06
IT: 800
loss_text_ce: 2.78
loss_mel_ce: 0.68

Some of the generation settings I tried:
(8-word prompt)
Voice Chunks: 0, 1, 16, 64
Use Original Latents Method (AR): checked and unchecked
Use Original Latents Method (Diffusion): checked and unchecked
Candidates: 1, 6 (changing this at all resulted in a cuda error, and I couldn't figure out how to fix it)
Preset: Ultra Fast and High Quality
Samples: 16, 256, 512
Iterations: 30, 128, 256, 512
Temperature: 0.2, 0.8
Show Experimental Settings: checked
Half Precision: unchecked
Condition-Free: checked
Pause Size: 8
Diffusion Samplers: DDIM
Run the outputted audio through RVC: checked and unchecked (this crashed the interface, so it's a whole other post...)
CVVP Weight: 0
Top P: 0.8
Diffusion Temperature: 1
Length Penalty: 1
Repetition Penalty: 2
Conditioning-Free K: 2

**Comment by GregMuck99** [comment]
I have worked with Jason's version 2 package almost every day since he released it and have had good success. I am guessing, but I suspect the problem is in your sample set, which is generated when you click "transcribe and process". If your audio is one long line of speech, this step will parse/split your 5 minute audio file into shorter clips, then attempt to transcribe each of those clips. This may not provide a good model if your audio is split in the middle of a word(s) or phenome. To check this, look in the TRAINING folder of one of your runs. Open the subfolder that has the name you chose when you created the subfolder under VOICES (which holds your original 5 minute audio file). When you open this TRAINING folder you will see several files and another subfolder named AUDIO. This AUDIO folder should contain a large number of little wav files that were written into the folder when you clicked "transcribe and process". Copy and paste this folder to your desktop so you can open each wav file and listen to it.

Next, copy and paste the train.txt and validation.txt files to you desktop. When you examine the lines in these text files, they will be the WHISPER transcription of each little wav file in the TRAINING\AUDIO folder. To train a good model, the words you hear in each little wav file must be correctly transcribed in the train.txt and validate.txt files. If they don't match the TTS voice model won't know what sounds to use for a given text prompt. I can write another long explanation of how to fix this, but before I waste your time and mine, let me know if the little wav files and their corresponding transcriptions match.

If the model is flawed, no combination of settings under GENERATE will fix it. I have been in that rabbit hole and I feel your pain.
